extensions/config-yaml/runtime/src/main/resources/META-INF/services/org.eclipse.microprofile.config.spi.ConfigSource

```diff
@@ -0,0 +1,4 @@
+io.quarkus.yaml.configuration.runtime.MicroprofileConfigJarYamlConfigSource
+io.quarkus.yaml.configuration.runtime.ApplicationJarYamlConfigSource
+io.quarkus.yaml.configuration.runtime.MicroprofileConfigFileSystemYamlConfigSource
+io.quarkus.yaml.configuration.runtime.ApplicationFileSystemYamlConfigSource
```

Comment:
Not sure I like these many ConfigSources for just reading YAML files, but maybe I am missing how the Config API works

Reply:
Ah you could implement a ConfigSourceProvider instead: https://github.com/eclipse/microprofile-config/blob/master/api/src/main/java/org/eclipse/microprofile/config/spi/ConfigSourceProvider.java#L51

Reply:
I added a `ConfigSourceProvider` implementation but it doesn't really change the fact that we need 4 config sources to cover all YAML cases.

Reply:
You don't really need 4 classes if you pass an URL and the ordinal in the YamlConfigSource class constructor, just saying :)

Reply:
Yeah, I still need to think about that, I didn't forget your previous comment ;)